app/models/contact_info.rb

```diff
@@ -48,6 +48,11 @@ def init_confirmation_code!
     self.confirmation_code ||= TokenMaker.contact_info_confirmation_code
   end
 
+  def set_confirmation_pin_code
+    self.confirmation_pin ||= TokenMaker.contact_info_confirmation_pin
+    self.confirmation_code ||= TokenMaker.contact_info_confirmation_code
+  end
```

Comment:
Maybe name this `reset_` and make it always change the pin/code.
Having to manually clear the pin and code before calling this in ChangeSignupEmail is weird.

Reply:
Defined it, `reset_confirmation_pin_code`